### PR TITLE
Add basic support for OS X

### PR DESCRIPTION
**Pull request by patternoia**
Hi, here is a product of my short hacking around your code.
I will not have much time to support it further, but this could be a good foundation to possibly further extend the code to support linux/other posix compatible systems.

Cheers,
pat

### CODE REVIEW

**Comment by remyroy** [comment]
Awesome! I'll try to check it out shortly and see if I could release a OSX version soon.

File: cddagl/ui.py [hunk 1]
```diff
@@ -378,7 +470,7 @@ def check_new_launcher_version(self):
     def lv_http_finished(self):
         self.lv_html.seek(0)
         document = html5lib.parse(self.lv_html, treebuilder='lxml',
-            encoding='utf8', namespaceHTMLElements=False)
+            namespaceHTMLElements=False)
```
Comment (patternoia) [inline cddagl/ui.py]: On my python deployment (Python 3.5.1 from Homebrew) there is no such parameter as 'encoding'

File: cddagl/ui.py [hunk 2]
```diff
@@ -497,7 +589,7 @@ def __init__(self):
 
                 try:
                     self.pipe = SimpleNamedPipe('cddagl_instance')
-                except (OSError, PyWinError):
+                except (OSError):
```
Comment (patternoia) [inline cddagl/ui.py]: PyWinError should be generalized to some other which is platform independent. I excluded it for now.

File: cddagl/ui.py [hunk 3]
```diff
@@ -2208,6 +2351,27 @@ def run(self):
 
                     self.completed.emit()
 
+            class TestingImgThread(QThread):
```
Comment (patternoia) [inline cddagl/ui.py]: This check could be skipped, since the image verification happens during the mounting. But I left it in place for consistency.

**Comment by patternoia** [comment]
It would be great to refactor and segment the program to divide single ~8k lines `ui.py` file into smaller blocks. I really had some problems with it

File: cddagl/ui.py [hunk 4]
```diff
@@ -122,11 +161,64 @@ def arstrip(value):
...
+    return platform.system() == 'Windows'
+
+def is_linux():
+    return platform.system() == 'Linux'
+
+def is_posix():
+    return os.name == 'posix'
+
+def is_64_system():
+    return sys.maxsize > 2**32
```
Comment (remyroy) [inline cddagl/ui.py]: This will fail detecting a 64-bit OS that runs a 32-bit version of Python. This needs to be changed so it can detect the OS bitness.

Comment (patternoia) [inline cddagl/ui.py]: Agree, platform.architecture()[0] gives more reliable result according to http://stackoverflow.com/questions/1842544/how-do-i-detect-if-python-is-running-as-a-64-bit-application

Comment (remyroy) [inline cddagl/ui.py]: platform.architecture()[0] will also fail to detect a 64-bit OS that runs a 32-bit version of Python. On my Windows 64-bit, here is what I get:

```python
>>> platform.architecture()[0]
'32bit'
```

**Comment by remyroy** [comment]
This is related to #73.

**Comment by patternoia** [comment]
Should close this, since it is too inmature and stale to be merged in near future.

**Pull request closed without merging**